compiler/setup.py

```diff
@@ -464,6 +464,7 @@ def find_git_submodule_revision(submodule_path):
     install_requires=[
         "numpy",
         "PyYAML",
+        "sympy",
```

Comment:
Looks like this new dep is for the `fx_importer` in torch-mlir. That should have at least been documented in the PR description.

This should probably go in `extras_require` for a `iree-compiler[torch]` package, not the general requirements here. The dep seems relatively small, but I want to keep the core project's dependencies minimal.